README.md

```diff
@@ -183,6 +183,80 @@ By making a contribution to this project, I certify that:
+
+Due to the existing layering, modules and applications are sensitive not only to changes in the Core Library API, but the Application Binary Interface and dependendencies as well.
+
+Any API *addition* to either the Node.js Core Library API or Application Binary Interface must result in a *semver-minor* version increase.
+
+Any *backwards incompatible* change to either the Node.js Core Library API or Application Binary Interface must result in a *semver-major* version increase.
+
+Issue: Should any modification to the ABI or Dependencies that requires module or application developers to recompile force a *semver-major* change?
+
+APIs and default behaviors in the Node.js Core Library, Application Binary Interface, Dependencies and nan must not change within LTS Releases unless the change is required to address a critical security update.  
+
+Note that default or assumed behaviors and values are exposed via the API, ABI or Dependencies are considered to be part of the API. A change in a default value, for instance, counts as an API change as modules and applications may be written to assume certain defaults and could be broken.
+
+### Dependency Stability
+
+#### JavaScript Support (V8)
+
+Node.js will adopt new V8 releases as quickly as practically feasible. For LTS Releases, the version of V8 shipped must be supported and fully operational on all platforms officially targeted for support.
+
+When V8 ships a breaking change to their C++ API that can be handled by `nan` there will be a *semver-minor* version increase.
+
+When V8 ships a breaking change to their C++ API that cannot be handled by `nan` there will be a *semver-major* version increase.
+
+When new features in the JavaScript language are introduced by V8, there will be a *semver-minor* version increase. TC39 has stated clearly that no backwards incompatible changes will be made to the language so it is appropriate to increase the *minor* rather than the *major*.
+
+Pull Requests that introduce post-ES5 mechanisms into the Node.js Core Library API require TSC review and approval. If landed, such changes must result in a *semver-major* version increase. However, Pull Requests may introduce post-ES5 mechanisms into the *internal* JavaScript implementation of the Core Library API so long as those mechanisms do not "bleed out" through the API.
```

Comment:
This is for things like adding promises into the core API. It's been a
significant enough of an issue historically that such changes warrant
additional review. If there is consensus on the TSC that this isn't a
concern, then this can be dropped but I feel it at least warrants
discussion before doing so.
On Apr 7, 2015 2:15 PM, "Jeremiah Senkpiel" <notifications@github.com>
wrote:

> In README.md
> <https://github.com/jasnell/dev-policy/pull/21#discussion_r27921267>:
>
> > +
> > +Note that default or assumed behaviors and values are exposed via the API, ABI or Dependencies are considered to be part of the API. A change in a default value, for instance, counts as an API change as modules and applications may be written to assume certain defaults and could be broken.
> > +
> > +### Dependency Stability
> > +
> > +#### JavaScript Support (V8)
> > +
> > +Node.js will adopt new V8 releases as quickly as practically feasible. For LTS Releases, the version of V8 shipped must be supported and fully operational on all platforms officially targeted for support.
> > +
> > +When V8 ships a breaking change to their C++ API that can be handled by `nan` there will be a *semver-minor* version increase.
> > +
> > +When V8 ships a breaking change to their C++ API that cannot be handled by `nan` there will be a *semver-major* version increase.
> > +
> > +When new features in the JavaScript language are introduced by V8, there will be a *semver-minor* version increase. TC39 has stated clearly that no backwards incompatible changes will be made to the language so it is appropriate to increase the *minor* rather than the *major*.
> > +
> > +Pull Requests that introduce post-ES5 mechanisms into the Node.js Core Library API require TSC review and approval. If landed, such changes must result in a *semver-major* version increase. However, Pull Requests may introduce post-ES5 mechanisms into the *internal* JavaScript implementation of the Core Library API so long as those mechanisms do not "bleed out" through the API.
>
> Isn't this harmful? Why would these be semver-major? It's not like these
> would need to be flagged or use external libs...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jasnell/dev-policy/pull/21/files#r27921267>.
>